packages/eslint-plugin/tests/rules/only-throw-error.test.ts

```diff
@@ -139,6 +139,58 @@ function fun<T extends Error>(t: T): void {
   throw t;
 }
     `,
+    {
+      code: `
+throw undefined;
+      `,
+      options: [
+        {
+          allowThrowing: [{ from: 'lib', name: 'undefined' }],
+          allowThrowingAny: false,
+          allowThrowingUnknown: false,
```

Comment:
[Style] These default to `false`, no need to explicitly set them:

```suggestion
```

(here and below)

Reply:
@JoshuaKGoldberg The default options in documentation and code are different. In my opinion, the documentation should be changed.

https://github.com/typescript-eslint/typescript-eslint/blob/4af866a6642502117c32b49e64d7012a027c42f5/packages/eslint-plugin/src/rules/only-throw-error.ts#L58-L59

https://github.com/typescript-eslint/typescript-eslint/blob/4af866a6642502117c32b49e64d7012a027c42f5/packages/eslint-plugin/docs/rules/only-throw-error.mdx?plain=1#L117-L118

Reply:
aff345c